src/hera/workflows/parameter.py

```diff
@@ -93,4 +94,15 @@ def as_output(self) -> _ModelParameter:
         )
 
 
-__all__ = ["Parameter"]
+class InputParameterAnnotation(BaseModel):
```

Comment:
Ideally would like this to be the same as the Parameter class so that we don't diverge too much. We can also add some annotation specific fields that don't get exported to the hera.models.Parameters.

One such field would be `alias`